src/agent/src/device.rs

```diff
@@ -438,204 +459,174 @@ fn scan_scsi_bus(scsi_addr: &str) -> Result<()> {
+    let devidx = DevIndex::new(spec);
 
-    if !Path::new(vm_path).exists() {
-        return Err(anyhow!("vm_path:{} doesn't exist", vm_path));
-    }
+    for fixup in fixups {
+        let major_id: c_uint;
+        let minor_id: c_uint;
 
-    let meta = fs::metadata(vm_path)?;
-    let dev_id = meta.rdev();
-    unsafe {
-        major_id = major(dev_id);
-        minor_id = minor(dev_id);
-    }
+        // If no container_path is provided, we won't be able to match and
+        // update the device in the OCI spec device list. This is an error.
+        if fixup.old.is_empty() {
+            return Err(anyhow!("Host path cannot empty for device"));
+        }
 
-    info!(
-        sl!(),
-        "update_spec_device(): vm_path={}, major: {}, minor: {}\n", vm_path, major_id, minor_id
-    );
+        let linux = spec
+            .linux
+            .as_mut()
+            .ok_or_else(|| anyhow!("Spec didn't container linux field"))?;
 
-    if let Some(idxdata) = devidx.0.get(host_path) {
-        let dev = &mut linux.devices[idxdata.idx];
-        let host_major = dev.major;
-        let host_minor = dev.minor;
+        if !Path::new(&fixup.vm).exists() {
+            return Err(anyhow!("vm_path:{} doesn't exist", fixup.vm));
+        }
 
-        dev.major = major_id as i64;
-        dev.minor = minor_id as i64;
-        dev.path = final_path.to_string();
+        let meta = fs::metadata(&fixup.vm)?;
+        let dev_id = meta.rdev();
+        unsafe {
+            major_id = major(dev_id);
+            minor_id = minor(dev_id);
+        }
 
         info!(
             sl!(),
-            "change the device from path: {} major: {} minor: {} to vm device path: {} major: {} minor: {}",
-            host_path,
-            host_major,
-            host_minor,
-            dev.path,
-            dev.major,
-            dev.minor,
+            "update_spec_device(): vm_path={}, major: {}, minor: {}\n", fixup.vm, major_id, minor_id
         );
 
-        // Resources must be updated since they are used to identify
-        // the device in the devices cgroup.
-        for ridx in &idxdata.residx {
-            // unwrap is safe, because residx would be empty if there
-            // were no resources
-            let res = &mut linux.resources.as_mut().unwrap().devices[*ridx];
-            res.major = Some(major_id as i64);
-            res.minor = Some(minor_id as i64);
+        if let Some(idxdata) = devidx.0.get(&fixup.old) {
+            let dev = &mut linux.devices[idxdata.idx];
+            let old_major = dev.major;
+            let old_minor = dev.minor;
+
+            dev.major = major_id as i64;
+            dev.minor = minor_id as i64;
+            dev.path = fixup.new.to_string();
 
             info!(
                 sl!(),
-                "set resources for device major: {} minor: {}\n", major_id, minor_id
+                "change the device from path: {} major: {} minor: {} to vm device path: {} major: {} minor: {}",
```

Comment:
I've removed the other one.

Reply:
Wait... no I haven't.  That's not in scope for the patches I've written so far, since it's basically just indenting the existing messages in an additional block.